mm2src/coins/nft.rs

```diff
@@ -158,24 +162,68 @@ pub async fn get_nft_transfers(ctx: MmArc, req: NftTransfersReq) -> MmResult<Nft
     let nft_ctx = NftCtx::from_ctx(&ctx).map_to_mm(GetNftInfoError::Internal)?;
     let _lock = nft_ctx.guard.lock().await;
 
-    let storage = NftStorageBuilder::new(&ctx).build()?;
+    let storage = nft_ctx.lock_db().await?;
     for chain in req.chains.iter() {
         if !NftTransferHistoryStorageOps::is_initialized(&storage, chain).await? {
             NftTransferHistoryStorageOps::init(&storage, chain).await?;
         }
     }
     let mut transfer_history_list = storage
-        .get_transfer_history(req.chains, req.max, req.limit, req.page_number, req.filters)
+        .get_transfer_history(req.chains.clone(), req.max, req.limit, req.page_number, req.filters)
         .await?;
     if req.protect_from_spam {
         for transfer in &mut transfer_history_list.transfer_history {
             protect_from_history_spam_links(transfer, true)?;
         }
     }
+    process_transfers_confirmations(&ctx, req.chains, &mut transfer_history_list).await?;
     drop_mutability!(transfer_history_list);
     Ok(transfer_history_list)
 }
 
+async fn process_transfers_confirmations(
+    ctx: &MmArc,
+    chains: Vec<Chain>,
+    history_list: &mut NftsTransferHistoryList,
+) -> MmResult<(), TransferConfirmationsError> {
+    let mut blocks_map = HashMap::with_capacity(chains.len());
+    for chain in chains.into_iter() {
+        let coin_enum = lp_coinfind_or_err(ctx, chain.to_ticker()).await?;
+        let current_block = match coin_enum {
+            MmCoinEnum::EthCoin(eth_coin) => current_block_impl(eth_coin).await?,
+            _ => {
+                return MmError::err(TransferConfirmationsError::CoinDoesntSupportNft {
+                    coin: coin_enum.ticker().to_owned(),
+                })
+            },
+        };
+        blocks_map.insert(chain.to_ticker(), current_block);
+    }
```

Comment:
We should aim to do such operations concurrently instead of sequentially.

Reply:
Thanks for the note, done

Reply:
Please consider this diff, it's a bit simpler
```diff
Index: mm2src/coins/nft.rs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/mm2src/coins/nft.rs b/mm2src/coins/nft.rs
--- a/mm2src/coins/nft.rs	(revision 03b01c55249bdb18443ea9fc2d637a5ce3129456)
+++ b/mm2src/coins/nft.rs	(date 1699450209742)
@@ -26,7 +26,7 @@
 use crypto::StandardHDCoinAddress;
 use ethereum_types::{Address, H256};
 use futures::compat::Future01CompatExt;
-use futures::future::join_all;
+use futures::future::try_join_all;
 use mm2_err_handle::map_to_mm::MapToMmResult;
 use mm2_net::transport::send_post_request_to_uri;
 use mm2_number::{BigDecimal, BigUint};
@@ -187,8 +187,6 @@
     chains: Vec<Chain>,
     history_list: &mut NftsTransferHistoryList,
 ) -> MmResult<(), TransferConfirmationsError> {
-    let mut blocks_map = HashMap::with_capacity(chains.len());
-
     let futures = chains.into_iter().map(|chain| async move {
         let ticker = chain.to_ticker();
         let coin_enum = lp_coinfind_or_err(ctx, ticker).await?;
@@ -202,11 +200,7 @@
             }),
         }
     });
-    let results = join_all(futures).await;
-    for result in results {
-        let (ticker, current_block) = result?;
-        blocks_map.insert(ticker, current_block);
-    }
+    let blocks_map = try_join_all(futures).await?.into_iter().collect::<HashMap<_, _>>();
 
     for transfer in history_list.transfer_history.iter_mut() {
         let current_block = match blocks_map.get(transfer.chain.to_ticker()) {
```

Reply:
done